Add Private location capabilities guide (draft) (#25052)
newrelic/docs-website: src/content/docs/synthetics/synthetic-monitoring/guides/private-location-capabilities.mdx

---
title: Private location capabilities
tags:
  - Synthetics
  - Synthetic monitoring
  - Private locations
metaDescription: 'A consolidated overview of what New Relic synthetic monitoring private locations can do: security, compliance, networking, scaling, and automation.'
freshnessValidatedDate: never
---

> A private location is a collection of **synthetics job managers (SJMs)** — containers you run in your own environment. Because the SJM only makes **outbound** calls to New Relic's collector (the "Horde" endpoint), you can monitor internal, firewalled, or otherwise private targets **without opening any inbound access** to your network. Test scripts, secrets, and traffic stay inside your infrastructure. This contrasts with New Relic's managed **public locations**, which run from New Relic-operated infrastructure around the world.

Reviewer: Why are non-UI words in the text boldfaced?

Reviewer 2: Removed all instances of non-UI words in boldface.

> A private location is a collection of **synthetics job managers (SJMs)** — containers you run in your own environment. Because the SJM only makes **outbound** calls to New Relic's collector (the "Horde" endpoint), you can monitor internal, firewalled, or otherwise private targets **without opening any inbound access** to your network. Test scripts, secrets, and traffic stay inside your infrastructure. This contrasts with New Relic's managed **public locations**, which run from New Relic-operated infrastructure around the world.

Reviewer: ```suggestion
A private location is a collection of **synthetics job managers (SJMs)** — containers you run in your own environment. Because SJM only makes **outbound** calls to New Relic's collector (the "Horde" endpoint), you can monitor internal, firewalled, or otherwise private targets **without opening any inbound access** to your network. Test scripts, secrets, and traffic stay inside your infrastructure. This contrasts with New Relic's managed **public locations**, which run from New Relic-operated infrastructure around the world.
```

> The following table summarizes what private locations support. Each capability is detailed in the sections below.

Reviewer: If the table is only listing the supported scenarios, why do we have the "Supported" column? Remove it.

Reviewer 2: Removed the column.

    * The SJM **runs on Docker, Podman, Kubernetes, and OpenShift** from a single image (`newrelic/synthetics-job-manager`).
    * **Per-check sandboxing:** For each scripted or browser check, the SJM spins up a fresh runtime container or pod, then tears it down, isolating runs from one another.
    * **Runtimes the SJM manages:**
      * `synthetics-ping-runtime` — ping monitors (lightweight)
      * `synthetics-node-api-runtime` — scripted API, broken links, certificate check
      * `synthetics-node-browser-runtime` — simple and scripted browser monitors (Chrome)
    * **Outbound only:** The SJM must reach the Horde endpoint (US `https://synthetics-horde.nr-data.net/`, EU `https://synthetics-horde.eu01.nr-data.net/`, JP `https://synthetics-horde.jp.nr-data.net/`). Test connectivity with:

      ```bash
      curl -X GET https://synthetics-horde.nr-data.net/synthetics/api/v1/ping
      ```

    For setup, see [Private locations overview](/docs/synthetics/synthetic-monitoring/private-locations/private-locations-overview-monitor-internal-sites-add-new-locations/) and [Install the job manager](/docs/synthetics/synthetic-monitoring/private-locations/install-job-manager/).

>     <Callout variant="important" title="You control the runtime version">

Reviewer: Why is this in a callout? The text is too long to be effective in a callout. Can't we have it as normal text?

    * **Docker:** Mount to `/var/lib/newrelic/synthetics/certs` read-only.
    * **Kubernetes:** Create a secret and reference it with `global.customCertificates.volume.secret.secretName`.

  </Collapser>
</CollapserGroup>

See [Configure custom CA certificates](/docs/synthetics/synthetic-monitoring/private-locations/job-manager-configuration/#custom-ca-certificates).

### mTLS client certificates [#mtls]

**Problem it solves:** Services that require the caller to present a client certificate reject your monitors outright, which typically pushes you into building and maintaining a proxy per service just to get coverage. Hostname-mapped client certificates remove that workaround entirely.

<CollapserGroup>
  <Collapser
    id="mtls-details"
    title="Set up mTLS client certificates"
  >

    Monitors present the correct certificate per hostname automatically, with no script changes.

    * Provide a `cert-map.json` mapping hostnames to PEM cert/key files (matching priority: exact → wildcard `*.domain.com` → `default` → no certificate).
    * **Docker:** Mount your client-cert directory to `/var/lib/newrelic/synthetics/client-cert` read-only.

>     * **Kubernetes:** Create a Secret containing `cert-map.json` and the cert/key files, then reference it with `global.clientCertificates.volume.secret.secretName`.

Reviewer: Why is secret capitalized? Are we referring to a feature name here?

### Non-root execution [#non-root]

**Problem it solves:** Container security standards in many organizations prohibit running workloads as root, which would otherwise block you from deploying the job manager at all. Non-root execution lets monitoring pass your platform security review.

<CollapserGroup>
  <Collapser
    id="non-root-details"
    title="Run the job manager as non-root"
  >

    The SJM runs as root by default (safe because execution is sandboxed), but it also supports running as a **non-root user**.

    * **Docker:** The user must be in the `docker` group (or use the Docker TCP socket via `DOCKER_HOST`) and have read/write access to mounted volumes, then run with `-u UID:GID`.
    * Runtime containers always run as UID `2000`.
    * **Kubernetes** exposes a `podSecurityContext` Helm value for `runAsUser`, `runAsNonRoot`, and `fsGroup`. Podman supports a fully rootless model.

  </Collapser>
</CollapserGroup>

See [Security, sandboxing, and running as non-root](/docs/synthetics/synthetic-monitoring/private-locations/install-job-manager/#run-non-root).

### Verified script execution (VSE) [#vse]

> **Problem it solves:** Scripted monitors are arbitrary code, so anyone with account access could run their scripts on your infrastructure inside your network. A passphrase you alone hold ensures only trusted scripts execute on your job manager.

Reviewer: ```suggestion
**Problem it solves:** Scripted monitors are arbitrary code, so anyone with account access can run their scripts on your infrastructure inside your network. A passphrase you alone hold ensures only trusted scripts execute on your job manager.
```

<CollapserGroup>
  <Collapser
    id="vse-details"
    title="Enable verified script execution"
  >

    You must set a passphrase before anyone can assign scripts to your location or add SJMs to it. Note that VSE applies only to scripted monitors. Non-scripted monitors (like simple ping) are exempt from entering the passphrase.

    * Set the passphrase with `VSE_PASSPHRASE` (Docker/Podman) or `synthetics.vsePassphrase` / `synthetics.vsePassphraseSecretName` (Kubernetes), then enable VSE in the location's **Edit** menu and enter the passphrase for each monitor.

    <Callout variant="caution">

>       New Relic **never stores** the VSE passphrase, and no one — including New Relic Support — can recover it. If it's lost, you must reset it on the SJM and re-enter it on every monitor assigned to the location.

Reviewer: ```suggestion
      New Relic **never stores** the VSE passphrase, and no one — including New Relic Support — can recover it. If it's lost, you must reset it on the SJM and reauthenticate every monitor assigned to the location.
```

  >

    Sharing is controlled per location, so you decide which accounts can send jobs to your infrastructure.

    * Setting `shared` to `true` makes the location usable by all accounts in your organization.
    * A location created on a parent account is usable by its child accounts. A location created on a child account stays private to that account.
    * You can't unshare a location while monitors in other accounts still use it — migrate those monitors first.

  </Collapser>
</CollapserGroup>

See [Private locations overview](/docs/synthetics/synthetic-monitoring/private-locations/private-locations-overview-monitor-internal-sites-add-new-locations/).

### Secure credentials [#secure-credentials]

**Problem it solves:** Testing authenticated journeys requires real credentials, but putting them in scripts exposes them to anyone who can read the script or the check results. An encrypted vault keeps secrets usable but unreadable, and rotating one value updates every monitor at once.

<CollapserGroup>
  <Collapser
    id="secure-credentials-details"
    title="Review how secure credentials work"
  >

    Reference stored secrets in your scripts with `$secure.MY_KEY`.

>     * Uses **AES-GCM 256-bit encryption at rest**, with AWS Key Management Service (KMS) managing the keys. You can never read values back — you can only reference them.

Reviewer: Use DNT tag for "AWS Key Management Service (KMS)"

Reviewer 2: Enclosed in DNT tag.

    * Customer obligations include a New Relic-approved account, Enterprise edition with Data Plus (or an approved alternative), agents and services configured to FedRAMP-designated endpoints, and use of FedRAMP-authorized features only. FedRAMP is inherited by child accounts.

  </Collapser>
</CollapserGroup>

See [FedRAMP](/docs/security/security-privacy/compliance/certificates-standards-regulations/fedramp/) and [FedRAMP-compliant endpoints](/docs/security/security-privacy/compliance/fedramp-compliant-endpoints/).

## Networking [#networking]

### Dual stack (IPv4 + IPv6) [#dual-stack]

**Problem it solves:** As networks adopt IPv6, monitors that can only reach IPv4 leave the IPv6 path unverified — a gap where users hit failures you never see. Dual stack validates both address families from one location.

<CollapserGroup>
  <Collapser
    id="dual-stack-details"
    title="Review dual stack requirements"
  >

>     There's no in-product toggle — the SJM inherits dual-stack capability from the host or cluster network, gated on minimum image versions:

Reviewer: ```suggestion
    There's no in-product toggle — SJM inherits dual-stack capability from the host or cluster network, gated on minimum image versions:
```

    * Job manager `519` or later, ping runtime `1.65.0` or later, and current Node API and browser runtime images — use the `latest` tag.

>     * **Docker:** The host and Docker daemon must have IPv6 enabled. **Kubernetes:** The cluster must be v1.20+ with dual stack enabled. A Podman guide is also available.

Reviewer: Should Kubernetes be on a new bullet?

Reviewer 2: Yes, added it in the bullet list.

    title="Map hostnames to IP addresses"
  >

    On Docker deployments, set `RUNTIME_EXTRA_HOSTS` on the job manager and it passes the mappings into every runtime container it spawns, so your scripts can use the same hostnames your applications use.

    * **Format:** Comma-separated `hostname:ip` pairs, for example `api.internal.example.com:10.0.0.1,svc.internal:10.0.0.2`.
    * Useful whenever an internal name isn't resolvable from the runtime — including mTLS targets referenced in `cert-map.json`.

  </Collapser>
</CollapserGroup>

See [mTLS client certificates](/docs/synthetics/synthetic-monitoring/private-locations/job-manager-configuration/#mtls-client-certificates).

## Extensibility [#extensibility]

### Custom Node modules [#custom-modules]

**Problem it solves:** Prevents unmonitored gaps and extra tool sprawl by allowing you to import third-party packages for protocols, cloud SDKs, or data formats missing from the default runtime.

<CollapserGroup>
  <Collapser
    id="custom-modules-details"
    title="Set up custom node modules"
  >

    Bundle your own npm packages (hosted or local) for use with `require()` in scripted API and browser monitors.

>     * Provide a directory with a root `package.json`, and the SJM runs `npm install` on startup.

Reviewer: ```suggestion
    * Provide a directory with a root `package.json`, and SJM runs `npm install` on startup.
```

    * Mount to `/var/lib/newrelic/synthetics/modules` (Docker/Podman, read/write). On Kubernetes, use a PersistentVolume (`ReadWriteMany` if shared across pods) via `global.customNodeModules.customNodeModulesPath`.

  </Collapser>
</CollapserGroup>

See [Custom node modules](/docs/synthetics/synthetic-monitoring/private-locations/job-manager-configuration/#custom-modules).

### User-defined variables [#user-defined-variables]

**Problem it solves:** Hardcoding environment-specific values into scripts means maintaining a near-duplicate script per environment and editing each one whenever something changes. Injecting configuration lets one script run everywhere.

<CollapserGroup>
  <Collapser
    id="user-defined-variables-details"
    title="Set up user-defined variables"
  >

    Access injected values in your scripts with `$env.USER_DEFINED_VARIABLES.MY_VARIABLE`.

>     * Set `USER_DEFINED_VARIABLES` (a JSON string) or mount a `user_defined_variables.json` file.

Reviewer: Avoid single bullet lists unless absolutely necessary

Reviewer 2: Done.

### Custom CA certificates [#custom-certs]

**Problem it solves:** Monitors can't verify internal HTTPS endpoints signed by your own certificate authority, so business-critical internal apps either go unmonitored or force you to maintain custom runtime images. Mounting your CA certificates makes those endpoints monitorable with TLS validation intact.

<CollapserGroup>
  <Collapser
    id="custom-certs-details"
    title="Set up custom CA certificates"
  >

    The SJM auto-detects mounted certificates and propagates them (read-only) to every runtime it spawns, so no custom runtime image is required.

    * **Format:** PEM, with a `.pem` extension, readable by UID `2000`. Only certificates with the `CA:TRUE` basic constraint are loaded, so leaf certificates are ignored.
    * **Docker:** Mount to `/var/lib/newrelic/synthetics/certs` read-only.
    * **Kubernetes:** Create a secret and reference it with `global.customCertificates.volume.secret.secretName`.

  </Collapser>
</CollapserGroup>

See [Configure custom CA certificates](/docs/synthetics/synthetic-monitoring/private-locations/job-manager-configuration/#custom-ca-certificates).

### mTLS client certificates [#mtls]

> **Problem it solves:** Services that require the caller to present a client certificate reject your monitors outright, which typically pushes teams into building and maintaining a proxy per service just to get coverage. Hostname-mapped client certificates remove that workaround entirely.

Reviewer 3: @newrelicsaurabh I've added little details which are missed here in this Request https://github.com/newrelicsaurabh/docs-website/pull/1/changes can you please add them to this PR

Author: Thanks for the call out. Incorporate the changes.

<CollapserGroup>
  <Collapser
    id="non-root-details"
    title="Run the job manager as non-root"
  >

    The SJM runs as root by default (safe because execution is sandboxed), but it also supports running as a **non-root user**.

    * **Docker:** The user must be in the `docker` group (or use the Docker TCP socket via `DOCKER_HOST`) and have read/write access to mounted volumes, then run with `-u UID:GID`.
    * Runtime containers always run as UID `2000`.
    * **Kubernetes** exposes a `podSecurityContext` Helm value for `runAsUser`, `runAsNonRoot`, and `fsGroup`. Podman supports a fully rootless model.

  </Collapser>
</CollapserGroup>

See [Security, sandboxing, and running as non-root](/docs/synthetics/synthetic-monitoring/private-locations/install-job-manager/#run-non-root).

### Verified script execution (VSE) [#vse]

**Problem it solves:** Scripted monitors are arbitrary code, so anyone with account access could run their scripts on your infrastructure inside your network. A passphrase you alone hold ensures only trusted scripts execute on your job manager.

<CollapserGroup>
  <Collapser
    id="vse-details"
    title="Enable verified script execution"
  >

>     A passphrase is required before anyone can assign scripts to your location or add SJMs to it.

Reviewer 4: can we Clarify here VSE applies only to scripted monitors (scripted_api, scripted_browser) and non-scripted types (like simple ping) are exempt from entering the passphrase.

Author: Thanks for the call out - incorporated the language

    * Set the passphrase with `VSE_PASSPHRASE` (Docker/Podman) or `synthetics.vsePassphrase` / `synthetics.vsePassphraseSecretName` (Kubernetes), then enable VSE in the location's **Edit** menu and enter the passphrase for each monitor.

    <Callout variant="caution">

      New Relic **never stores** the VSE passphrase, and no one — including New Relic Support — can recover it. If it's lost, you must reset it on the SJM and re-enter it on every monitor assigned to the location.

    </Callout>

  </Collapser>
</CollapserGroup>

See [Verified script execution for private locations](/docs/synthetics/synthetic-monitoring/private-locations/verified-script-execution-private-locations/).

### Shared locations [#shared-locations]

> **Problem it solves:** A private location created in one account isn't usable by other teams by default, so each team stands up its own job managers for the same network — duplicating infrastructure, cost, and maintenance. Sharing lets one location serve many accounts.

Reviewer 5: ```suggestion
**Problem it solves:** A private location created in one account isn't usable by other accounts by default, so each account has its dedicated job managers for the same network — duplicating infrastructure, cost, and maintenance. Sharing lets one location serve many accounts.
```

<CollapserGroup>
  <Collapser
    id="shared-locations-details"
    title="Share a location across accounts"
  >

>     Sharing is controlled per location, so you decide which teams can send jobs to your infrastructure.

Reviewer 5: ```suggestion
    Sharing is controlled per location, so you decide which accounts can send jobs to your infrastructure.
```

  >

    Set these variables on the job manager:

    * **Docker/Podman:** `HORDE_API_PROXY_HOST`, `HORDE_API_PROXY_PORT`, `HORDE_API_PROXY_USERNAME`, `HORDE_API_PROXY_PW`, and `HORDE_API_PROXY_ACCEPT_SELF_SIGNED_CERT`.
    * **Kubernetes:** `synthetics.apiProxyHost`, `synthetics.apiProxyPort`, `synthetics.hordeApiProxyUsername`, `synthetics.hordeApiProxyPw`, and `synthetics.hordeApiProxySelfSignedCert`.

    <Callout variant="tip">

      These variables govern SJM-to-New-Relic traffic. There's no documented general-purpose proxy variable for arbitrary monitor-target traffic.

    </Callout>

  </Collapser>
</CollapserGroup>

See [Job manager configuration](/docs/synthetics/synthetic-monitoring/private-locations/job-manager-configuration/#environment-variables).

### Custom hostnames [#custom-hostnames]

**Problem it solves:** Monitors can't reach an internal service whose hostname resolves only inside your network — or resolves to the wrong address inside the runtime container — so the check fails on DNS instead of on the thing you meant to test.

<CollapserGroup>
  <Collapser
    id="custom-hostnames-details"
    title="Map hostnames to IP addresses"
  >

>     Set `RUNTIME_EXTRA_HOSTS` on the job manager and it passes the mappings into every runtime container it spawns, so your scripts can use the same hostnames your applications use.

Reviewer 6: ```suggestion
    On Docker deployments, set `RUNTIME_EXTRA_HOSTS` on the job manager and it passes the mappings into every runtime container it spawns, so your scripts can use the same hostnames your applications use.
```

Reviewer 6: This value is preset only in Docker flow. Kubernetes and Podman does not have support.

## Capabilities at a glance 

The following table summarizes what private locations support. Each capability is detailed in the sections below.

| Capability | What it gives you | Supported |
| --- | --- | --- |
| Deploy anywhere (Docker/Podman/Kubernetes/OpenShift) | Run monitors inside your own infrastructure, behind the firewall |  Yes  |
| Custom CA certificates | Trust internal or private-PKI HTTPS endpoints |    Yes |
| mTLS client certificates | Present client certs to internal mTLS endpoints (`cert-map.json`) |  Yes  |
| Non-root execution | Run the SJM as a non-root user in hardened containers |  Yes  |
| Verified script execution (VSE) | Passphrase-gate who can run scripts on your location |  Yes  |

> | Secure credentials | Encrypted secret vault, redacted from results and logs |  Yes  |

Reviewer 2: @newrelicsaurabh, can the **What it gives you** column content for this row be reworded to "Store secrets in an encrypted vault, redacted from results and logs"..?

Author: incorporated the change

| FedRAMP Moderate | Use synthetic monitoring in FedRAMP-authorized accounts |  Yes  |
| Dual stack (IPv4 + IPv6) | Monitor both IPv4 and IPv6 endpoints |  Yes  |
| Outbound proxy | Reach New Relic through an HTTP/HTTPS proxy |  Yes  |
| Custom Node modules | Use your own npm packages in scripts |  Yes  |
| User-defined variables | Inject non-secret configuration into scripts |  Yes  |
| Horizontal and vertical scaling | Add SJMs (same key) or add CPU/RAM for throughput |  Yes  |
| Automation and infrastructure as code | Manage locations with NerdGraph and Terraform |  Yes  |
| Shared locations | Share one location across all accounts in an org |  Yes  |

> | Health and observability | Status events plus a prebuilt job manager dashboard |  Yes  |

Reviewer 2: Hi @newrelicsaurabh, Can this change be made:
Status events plus a prebuilt job manager dashboard → Track status events with a prebuilt job manager dashboard

Author: Incorporated the change

> | Kerberos authentication | Kerberos SSO for internal sites and APIs | Soon |

Reviewer 2: @newrelicsaurabh, Can this change be made:
Kerberos SSO for internal sites and APIs → Authenticate with Kerberos SSO for internal sites and APIs

Reviewer 2: @newrelicsaurabh, we shouldn't be mentioning the upcoming releases in our docs. In that context, we should be removing this row.

Author: removed this

<CollapserGroup>
  <Collapser
    id="mtls-details"
    title="Set up mTLS client certificates"
  >

    Monitors present the correct certificate per hostname automatically, with no script changes.

    * Provide a `cert-map.json` mapping hostnames to PEM cert/key files (matching priority: exact → wildcard `*.domain.com` → `default` → no certificate).
    * **Docker:** mount your client-cert directory to `/var/lib/newrelic/synthetics/client-cert` read-only.
    * **Kubernetes:** create a Secret containing `cert-map.json` and the cert/key files, then reference it with `global.clientCertificates.volume.secret.secretName`.
    * Because mTLS is mutual, the monitor must also trust the server. If the endpoint's certificate is issued by a private CA, mount that CA as a [custom CA certificate (#custom-ca-certificates) (Docker: `/var/lib/newrelic/synthetics/certs`; Kubernetes: `global.customCertificates.volume.secret.secretName`).

>     * If the mTLS hostnames aren't resolvable via DNS inside the runtime, map them with [custom hostnames](#custom-hostnames).

Author: [custom hostnames](#custom-hostnames) isn't rendering

Reviewer 2: @newrelicsaurabh, I believe this is rendering?

Author: yes, this is.

### Custom hostnames [#custom-hostnames]

**Problem it solves:** Monitors can't reach an internal service whose hostname resolves only inside your network — or resolves to the wrong address inside the runtime container — so the check fails on DNS instead of on the thing you meant to test.

<CollapserGroup>
  <Collapser
    id="custom-hostnames-details"
    title="Map hostnames to IP addresses"
  >

    On Docker deployments, set `RUNTIME_EXTRA_HOSTS` on the job manager and it passes the mappings into every runtime container it spawns, so your scripts can use the same hostnames your applications use.

    * **Format:** comma-separated `hostname:ip` pairs, for example `api.internal.example.com:10.0.0.1,svc.internal:10.0.0.2`.
    * Useful whenever an internal name isn't resolvable from the runtime — including mTLS targets referenced in `cert-map.json`.

  </Collapser>
</CollapserGroup>

See [mTLS client certificates](/docs/synthetics/synthetic-monitoring/private-locations/job-manager-configuration/#mtls-client-certificates).

## Extensibility [#extensibility]

### Custom Node modules [#custom-modules]

> **Problem it solves:** A check that needs a protocol, cloud SDK, or data format the built-in runtime doesn't ship would otherwise be impossible — leaving that system unmonitored or pushing the work to a separate tool. Installing your own packages means the platform covers whatever you can express in code.

Reviewer 2: @newrelicsaurabh, shall we replace "the platform covers whatever you can express in code" with a scoped description such as "...lets you add protocols, SDKs, or data formats the built-in runtime doesn't ship"?

Author: Made the edits

    * Set `USER_DEFINED_VARIABLES` (a JSON string) or mount a `user_defined_variables.json` file.

    <Callout variant="caution">

      User-defined variables are **not** sanitized from logs. Use [secure credentials](#secure-credentials) for anything sensitive.

    </Callout>

  </Collapser>
</CollapserGroup>

See [User-defined variables for scripted monitors](/docs/synthetics/synthetic-monitoring/private-locations/job-manager-configuration/#user-defined-vars).

## Scaling and operations [#scaling]

**Problem it solves:** A single job manager has a fixed throughput ceiling, so as your monitor count grows the queue backs up and checks run late or not at all — silently reducing coverage. Scaling out adds capacity and failover.

<CollapserGroup>
  <Collapser
    id="scaling-details"
    title="Review sizing, scaling, and operations"
  >

    Private locations scale to match your monitoring load:

    * **Job types:** *Heavyweight* jobs (simple and scripted browser, scripted API) use roughly one CPU core per concurrent job. *Lightweight* jobs (ping) run on a worker-thread pool rather than a full core each.

>     * **Throughput per SJM:** Approximately 15 heavyweight jobs per minute and 75 ping checks per minute, by design.

Reviewer 7: Can we add the comupation formula link about how much load they might get for their monitor. 

For referrence https://docs.newrelic.com/docs/synthetics/synthetic-monitoring/private-locations/job-manager-configuration/#docker-podman-sizing

Author: Links added

<CollapserGroup>
  <Collapser
    id="automation-details"
    title="Review the API and Terraform options"
  >

    You can create and manage private locations programmatically:

    * **NerdGraph API:** `syntheticsCreatePrivateLocation`, `syntheticsUpdatePrivateLocation`, `syntheticsPurgePrivateLocationQueue`, and `syntheticsDeletePrivateLocation`, plus the secure-credential mutations. Creating a location returns a `guid`.
    * **Terraform:** The `newrelic_synthetics_private_location` resource (required `name` and `description`; optional `shared` and `verified_script_execution`). It exports the `key` you feed to your SJMs, so you can provision the location and wire up the runtime in one pipeline.

  </Collapser>
</CollapserGroup>

See [Manage private locations with NerdGraph](/docs/apis/nerdgraph/examples/synthetics-api/private-locations/) and the [Terraform provider docs](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/resources/synthetics_private_location).

> ## Coming soon [#coming-soon]

Reviewer 2: @newrelicsaurabh, we shouldn't be mentioning the upcoming releases in our docs. Having said that, we should be removing the "Coming soon" section entirely.

Author: Why waste an opportunity to tell what's in pipeline when we are consciously directing eyeballs to this page? This is sitting as a call out and not part of docs with references to current capabilities?
